### PR TITLE
Picture feature

### PR DESCRIPTION
**Pull request by knivets**

### CODE REVIEW

File: signup/models.py [hunk 1]
```diff
@@ -235,6 +235,8 @@ class Contact(models.Model):
         " followed by last name)"))
     nick_name = models.CharField(_("Nick name"), max_length=60, blank=True,
         help_text=_("Short casual name used to address the contact"))
+    picture = models.CharField(_("Profile picture"), max_length=300, blank=True,
```
Comment (smirolo) [inline signup/models.py]: I believe Django has a `models.URLField` that could be used here. Maybe. If it accepts s3:// and file:// prefixes.

File: signup/models.py [hunk 2]
```diff
@@ -285,6 +287,15 @@ def verification_key_expired(self):
                (start_at + expiration_date <= datetime_or_now())
     verification_key_expired.boolean = True
 
+    @property
+    def picture_url(self):
```
Comment (smirolo) [inline signup/models.py]: I don't understand the need for this method. Is the full URL not stored in the `picture` field?

File: signup/static/js/djaodjin-signup-vue.js [hunk 3]
```diff
@@ -120,8 +123,52 @@ var app = new Vue({
...
                 showErrorMessages(resp);
             });
+        },
+        getContact: function(cb){
+            var vm = this;
+            $.ajax({
+                method: 'GET',
+                url: djaodjinSettings.urls.user.api_contact,
+            }).done(function(res) {
+                console.log(res)
```
Comment (smirolo) [inline signup/static/js/djaodjin-signup-vue.js]: `console.log` statement shouldn't be committed. They have a tendency to break code when run on IE.

File: signup/views/users.py [hunk 4]
```diff
@@ -77,11 +77,20 @@ def get_context_data(self, **kwargs):
         setattr(context['user'], 'full_name', context['user'].get_full_name())
         # URLs for user
         if is_authenticated(self.request):
+            try:
+                contact = Contact.objects.get(user=self.object)
+            except Contact.DoesNotExist:
+                # TODO not sure how to handle this
```
Comment (smirolo) [inline signup/views/users.py]: See Django `get_or_create` method on models.

**Review by smirolo — APPROVED** [review]

File: signup/models.py (hunk 1 above)
Comment (knivets) [reply signup/models.py]: That's true, but I'm not actually storing the string in a valid URL format, the picture string is a concatenation of region, bucket and a file name. I can store it as a valid URL, this way we won't need the `picture_url` method.

File: signup/views/users.py (hunk 4 above)
Comment (knivets) [reply signup/views/users.py]: OK, but is it ok that I'm automatically creating a `Contact` object each time it doesn't exist for a current user?

Comment (smirolo) [reply signup/views/users.py]: It is OK, also something we do in other parts of the code base if I recall correctly.

**Pull request merged**